dirty_cat/gamma_poisson_factorization.py

```diff
@@ -0,0 +1,600 @@
+"""
+Online Gamma-Poisson factorization of string arrays.
+The principle is as follows:
+    1. Given an input string array X, we build its bag-of-n-grams
+       representation V (n_samples, vocab_size).
+    2. Instead of using the n-grams counts as encodings, we look for low-
+       dimensional representations by modeling n-grams counts as linear
+       combinations of topics V = HW, with W (n_topics, vocab_size) the topics
+       and H (n_samples, n_topics) the associated activations.
+    3. Assuming that n-grams counts follow a Poisson law, we fit H and W to
+       maximize the likelihood of the data, with a Gamma prior for the
```

Comment:
Is it better to say " to maximize their likelihood on the data" instead?

Reply:
I'd say "likelihood of the data", because we tune H and W so that the observed data is the most plausible.